cellmlmanip/parser.py

```diff
@@ -0,0 +1,195 @@
+"""
+This module contains the CellML parser. It reads CellML model and stores model information in a
+CellML Model class. MathML equations are translated to Sympy. RDF is handled by RDFLib.
+"""
+import itertools
+from enum import Enum
+from typing import Dict
+
+from lxml import etree
+
+from cellmlmanip import mathml2sympy
+from cellmlmanip.model import SYMPY_SYMBOL_DELIMITER, Component, Model
+
+
+class XmlNs(Enum):
+    """
+    Standard namespaces present in CellML documents
+    """
+    CELLML = 'http://www.cellml.org/cellml/1.0#'
+    CMETA = 'http://www.cellml.org/metadata/1.0#'
+    MATHML = 'http://www.w3.org/1998/Math/MathML'
+    RDF = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#'
+
+
+class Parser(object):
+    """Handles parsing of CellML files
+    """
+
+    @staticmethod
+    def with_ns(ns_enum, name):
+        """Returns an ElementTree-friendly name with namespace in brackets"""
+        return u'{%s}%s' % (ns_enum.value, name)
+
+    def __init__(self, filepath: str) -> None:
+        """Initialise an instance of Parser
+
+        :param filepath: the full filepath to the CellML model file
+        """
+        self.filepath: str = filepath
+        self.model: Model = None
+
+    def parse(self) -> Model:
+        """The main method that reads the XML file and extract the relevant parts of CellML model
+        definition. Parser class should have been instantiated with the filepath.
+
+        :return: a Model class holding CellML model definition, reading for manipulation
+        """
+        tree = etree.parse(self.filepath)
+
+        # <model> root node - initialise the model object
+        model_xml = tree.getroot()
+        self.model = Model(model_xml.get(Parser.with_ns(XmlNs.CMETA, u'id')))
+
+        # handle the child elements of <model>
+        self.__add_units(model_xml)
+        self.__add_components(model_xml)
+        self.__add_relationships(model_xml)
+        self.__add_rdf(model_xml)
+        self.__add_connection(model_xml)
+
+        return self.model
+
+    def __add_rdf(self, element: etree.Element):
```

Comment:
Why 2 leading underscores instead of 1? Normally this is reserved for python built-ins. (Also, 1 underscore already indicates hidden, why is that not enough?)

Reply:
I've switched to use single underscore